utils/ixdtf/src/parsers/records.rs

```diff
@@ -28,13 +29,14 @@ pub struct IsoParseRecord<'a> {
 
 #[non_exhaustive]
 #[derive(Debug, Clone, PartialEq)]
+/// A record of an annotation.
 pub struct Annotation<'a> {
     pub critical: bool,
     pub key: &'a str,
     pub value: &'a str,
 }
 
-#[non_exhaustive]
+#[allow(clippy::exhaustive_structs)]
```

Comment:
Nit: `#[allow(clippy::exhaustive_structs)]` should have a comment on the same line with a brief justification, such as

```suggestion
#[allow(clippy::exhaustive_structs)] // IXDTF defines Date as having three fields
```